### PR TITLE
update dependencies + lint updates

### PR DESCRIPTION
**Pull request by johnbrett**

### CODE REVIEW

**Comment by johnbrett** [comment]
Updating dependencies to fix https://github.com/hapijs/hapi/issues/3359

File: package.json [hunk 1]
```diff
@@ -14,10 +14,14 @@
   },
   "dependencies": {
     "b64": "3.x.x",
-    "boom": "3.x.x",
+    "boom": "^4.1.0",
+    "code": "^4.0.0",
```
Comment (DavidTPate) [inline package.json]: This dependency is only used in `test/index.js` shouldn't it be a `devDependency`?

File: package.json [hunk 2]
```diff
@@ -14,10 +14,14 @@
...
     "content": "3.x.x",
+    "form-data": "^2.1.1",
```
Comment (DavidTPate) [inline package.json]: This dependency is only used in `test/index.js` shouldn't it be a `devDependency`?

File: package.json [hunk 3]
```diff
@@ -14,10 +14,14 @@
...
     "hoek": "4.x.x",
-    "nigel": "2.x.x"
+    "lab": "^11.1.0",
```
Comment (DavidTPate) [inline package.json]: This dependency is only used in `test/index.js` shouldn't it be a `devDependency`?

File: package.json [hunk 4]
```diff
@@ -14,10 +14,14 @@
...
+    "nigel": "2.x.x",
+    "wreck": "^10.0.0"
```
Comment (DavidTPate) [inline package.json]: This dependency is only used in `test/index.js` shouldn't it be a `devDependency`?

**Comment by cjihrig** [comment]
Can you follow the existing format for modules in `package.json` of using `n.x.x` instead of the `^`. The exception should be any non-hapijs modules (like `form-data`), which I'd like to have pinned at an exact version.

Also, some of the modules now exist in both `dependencies` and `devDependencies`, which is wrong.

**Comment by johnbrett** [comment]
yep - sorry I'd used `--save` for each of them without thinking. Will update now.

File: package.json (hunk 1 above)
Comment (johnbrett) [inline package.json]: fixed - thanks!

File: package.json (hunk 2 above)
Comment (johnbrett) [inline package.json]: fixed - thanks!

File: package.json (hunk 3 above)
Comment (johnbrett) [inline package.json]: fixed - thanks!

File: package.json (hunk 4 above)
Comment (johnbrett) [inline package.json]: fixed - thanks!

File: package.json [hunk 5]
```diff
@@ -14,21 +14,17 @@
...
+    "nigel": "2.x.x"
   },
   "devDependencies": {
-    "code": "3.x.x",
-    "form-data": "0.2.x",
-    "lab": "10.x.x",
+    "code": "4.x.x",
+    "form-data": "2.x.x",
+    "lab": "11.x.x",
     "shot": "3.x.x",
```
Comment (cjihrig) [inline package.json]: Hm, doesn't look like `shot` is actually used anywhere.

Comment (johnbrett) [inline package.json]: Good spot

**Comment by cjihrig** [comment]
Thanks much!

**Comment by johnbrett** [comment]
No prob :)

**Pull request merged**